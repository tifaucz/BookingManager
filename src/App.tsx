import React from 'react';
import BookingForm from './components/BookingForm';
import BookingList from './components/BookingList'; 
import { BookingProvider } from './BookingContext';

const App: React.FC = () => {

  return (
    <BookingProvider>
      <div className="p-2 w-full h-full font-inter font-normal text-base text-[#00318d] bg-[#eae2dc] antialiased">
        <h1 className="text-2xl sm:text-3xl md:text-4xl lg:text-5xl xl:text-6xl" >Booking Manager</h1>
        <BookingForm />
        <BookingList />
      </div>
    </BookingProvider>
  );
};

export default App;