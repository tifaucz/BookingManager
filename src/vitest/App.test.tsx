import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import App from '../App';

describe('<App />', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});