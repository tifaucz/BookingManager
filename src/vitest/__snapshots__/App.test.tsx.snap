// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<App /> > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="p-2 w-full h-full font-inter font-normal text-base text-[#00318d] bg-[#eae2dc] antialiased"
  >
    <h1
      class="text-2xl sm:text-3xl md:text-4xl lg:text-5xl xl:text-6xl"
    >
      Booking Manager
    </h1>
    <div>
      <div
        class="flex justify-center my-4"
      >
        <div
          class="react-calendar react-calendar--selectRange"
        >
          <div
            class="react-calendar__navigation"
          >
            <button
              aria-label=""
              class="react-calendar__navigation__arrow react-calendar__navigation__prev2-button"
              disabled=""
              type="button"
            >
              «
            </button>
            <button
              aria-label=""
              class="react-calendar__navigation__arrow react-calendar__navigation__prev-button"
              disabled=""
              type="button"
            >
              ‹
            </button>
            <button
              aria-label=""
              class="react-calendar__navigation__label"
              style="flex-grow: 1;"
              type="button"
            >
              <span
                class="react-calendar__navigation__label__labelText react-calendar__navigation__label__labelText--from"
              >
                February 2024
              </span>
            </button>
            <button
              aria-label=""
              class="react-calendar__navigation__arrow react-calendar__navigation__next-button"
              type="button"
            >
              ›
            </button>
            <button
              aria-label=""
              class="react-calendar__navigation__arrow react-calendar__navigation__next2-button"
              type="button"
            >
              »
            </button>
          </div>
          <div
            class="react-calendar__viewContainer"
          >
            <div
              class="react-calendar__month-view"
            >
              <div
                style="display: flex; align-items: flex-end;"
              >
                <div
                  style="flex-grow: 1; width: 100%;"
                >
                  <div
                    class="react-calendar__month-view__weekdays"
                    style="display: flex; flex-wrap: nowrap;"
                  >
                    <div
                      class="react-calendar__month-view__weekdays__weekday react-calendar__month-view__weekdays__weekday--current"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    >
                      <abbr
                        aria-label="Monday"
                        title="Monday"
                      >
                        Mon
                      </abbr>
                    </div>
                    <div
                      class="react-calendar__month-view__weekdays__weekday"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    >
                      <abbr
                        aria-label="Tuesday"
                        title="Tuesday"
                      >
                        Tue
                      </abbr>
                    </div>
                    <div
                      class="react-calendar__month-view__weekdays__weekday"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    >
                      <abbr
                        aria-label="Wednesday"
                        title="Wednesday"
                      >
                        Wed
                      </abbr>
                    </div>
                    <div
                      class="react-calendar__month-view__weekdays__weekday"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    >
                      <abbr
                        aria-label="Thursday"
                        title="Thursday"
                      >
                        Thu
                      </abbr>
                    </div>
                    <div
                      class="react-calendar__month-view__weekdays__weekday"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    >
                      <abbr
                        aria-label="Friday"
                        title="Friday"
                      >
                        Fri
                      </abbr>
                    </div>
                    <div
                      class="react-calendar__month-view__weekdays__weekday react-calendar__month-view__weekdays__weekday--weekend"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    >
                      <abbr
                        aria-label="Saturday"
                        title="Saturday"
                      >
                        Sat
                      </abbr>
                    </div>
                    <div
                      class="react-calendar__month-view__weekdays__weekday react-calendar__month-view__weekdays__weekday--weekend"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    >
                      <abbr
                        aria-label="Sunday"
                        title="Sunday"
                      >
                        Sun
                      </abbr>
                    </div>
                  </div>
                  <div
                    class="react-calendar__month-view__days"
                    style="display: flex; flex-wrap: wrap;"
                  >
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                      disabled=""
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 29, 2024"
                      >
                        29
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                      disabled=""
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 30, 2024"
                      >
                        30
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                      disabled=""
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 31, 2024"
                      >
                        31
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      disabled=""
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="February 1, 2024"
                      >
                        1
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      disabled=""
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="February 2, 2024"
                      >
                        2
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                      disabled=""
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="February 3, 2024"
                      >
                        3
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                      disabled=""
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="February 4, 2024"
                      >
                        4
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      disabled=""
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="February 5, 2024"
                      >
                        5
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      disabled=""
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="February 6, 2024"
                      >
                        6
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      disabled=""
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="February 7, 2024"
                      >
                        7
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      disabled=""
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="February 8, 2024"
                      >
                        8
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      disabled=""
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="February 9, 2024"
                      >
                        9
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                      disabled=""
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="February 10, 2024"
                      >
                        10
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                      disabled=""
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="February 11, 2024"
                      >
                        11
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__tile--now react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="February 12, 2024"
                      >
                        12
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="February 13, 2024"
                      >
                        13
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="February 14, 2024"
                      >
                        14
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="February 15, 2024"
                      >
                        15
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="February 16, 2024"
                      >
                        16
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="February 17, 2024"
                      >
                        17
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="February 18, 2024"
                      >
                        18
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="February 19, 2024"
                      >
                        19
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="February 20, 2024"
                      >
                        20
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="February 21, 2024"
                      >
                        21
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="February 22, 2024"
                      >
                        22
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="February 23, 2024"
                      >
                        23
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="February 24, 2024"
                      >
                        24
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="February 25, 2024"
                      >
                        25
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="February 26, 2024"
                      >
                        26
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="February 27, 2024"
                      >
                        27
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="February 28, 2024"
                      >
                        28
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="February 29, 2024"
                      >
                        29
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="March 1, 2024"
                      >
                        1
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend react-calendar__month-view__days__day--neighboringMonth"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="March 2, 2024"
                      >
                        2
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend react-calendar__month-view__days__day--neighboringMonth"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="March 3, 2024"
                      >
                        3
                      </abbr>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex items-center space-x-4 p-2"
      >
        <input
          class="flex-1"
          name="name"
          placeholder="Name"
          value=""
        />
        <button
          class="px-4 py-2 border-blue text-blue"
          id="submit"
          type="button"
        >
          Add
        </button>
      </div>
    </div>
    <div
      class="flex items-center justify-between"
    />
  </div>
</DocumentFragment>
`;
